File: src/main/java/example/sort/service/pure/QuicksortService.java
package example.sort.service.pure;

public class QuicksortService implements example.sort.service.pure.SortService {
    private final example.sort.service.QuicksortService _otterop;

    private QuicksortService(example.sort.service.QuicksortService wrapped) {
        this._otterop = wrapped;
    }

    public QuicksortService() {
        this._otterop = new example.sort.service.QuicksortService();
    }

    public String[] sort(String[] array) {
        var _arrayArray = new otterop.lang.String[array.length];
        for (var i = 0; i < array.length; i++) {
            _arrayArray[i] = otterop.lang.String.wrap(array[i]);
        }
        var _array = otterop.lang.Array.wrap(_arrayArray);
        var retOtterop = _otterop.sort(_array);
        var ret = new String[retOtterop.size()];
        for (var i = 0; i < retOtterop.size(); i++) {
            var retI = retOtterop.get(i);
            ret[i] = retI.unwrap();
        }
        return ret;
    }

    public example.sort.service.QuicksortService unwrap() {
        return this._otterop;
    }

    public static example.sort.service.pure.QuicksortService wrap(example.sort.service.QuicksortService wrapped) {
        return new example.sort.service.pure.QuicksortService(wrapped);
    }
}

File: dotnet/src/Example/Sort/Service/Pure/QuicksortService.cs
namespace Example.Sort.Service.Pure
{
    public class QuicksortService : Example.Sort.Service.Pure.SortService
    {
        private Example.Sort.Service.QuicksortService otterop;

        private QuicksortService(Example.Sort.Service.QuicksortService wrapped)
        {
            this.otterop = wrapped;
        }
        public QuicksortService()
        {
            this.otterop = new Example.Sort.Service.QuicksortService();
        }

        public string[] Sort(string[] array)
        {
            var _arrayArray = new Otterop.Lang.String[array.Count()];
            for (var i = 0; i < array.Count(); i++)
            {
                _arrayArray[i] = Otterop.Lang.String.wrap(array[i]);
            }
            var _array = Otterop.Lang.Array.Wrap(_arrayArray);
            var retOtterop = otterop.Sort(_array);
            var ret = new string[retOtterop.Size()];
            for (var i = 0; i < retOtterop.Size(); i++)
            {
                var retI = retOtterop.Get(i);
                ret[i] = retI.unwrap();
            }
            return ret;
        }

        public Example.Sort.Service.QuicksortService unwrap()
        {
            return this.otterop;
        }

        public static Example.Sort.Service.Pure.QuicksortService wrap(Example.Sort.Service.QuicksortService wrapped)
        {
            return new Example.Sort.Service.Pure.QuicksortService(wrapped);
        }
    }

}
